xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code/src/main/java/org/xwiki/rendering/macro/code/CodeMacroLayout.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.code;
+
+import org.xwiki.rendering.code.layout.CodeLayoutHandler;
+
+/**
+ * Values allowed in the <code>layout</code> parameter of the code macro.
+ * 
+ * @version $Id$
+ * @since 11.3
+ */
+public enum CodeMacroLayout
```

Comment:
I guess you used an enum as an easy way to have all the possible layouts listed in the WYSIWYG but then it kind of defeat the purpose of using a public component since it makes impossible to provide more layouts without modifying this enum.

IMO either you make CodeLayoutHandler internal (what I would personally choose as a first version since I doubt adding more layouts than the 2 you implemented is that much of a need) or you move to String.

An alternative which is a bit more work is to introduce a custom type for this property (for example "CodeMacroLayout" but not an enum this time and just some String container) and provide a custom `org.xwiki.displayer.HTMLDisplayer` for this type which would allow you to control exactly what you want to have in he form.

Reply:
Good point. How would I go about making CodeLayoutHandler internal ? Remove it from DI entirely ?

Reply:
Just mean moving that class in an `*.internal.*` package (like the implementations).